### DISCUSSION TITLE
First Issue Friendly?

### DISCUSSION THREAD

**Discussion by kiridarivaki** [faker-js/faker#2746] [Q&A] [answered]
Hello everyone. I am a Computer Science student looking for an OSS to contribute to, for a course assignment. I have no experience in contributing to open source yet, i have knowledge of HTML, CSS and JS as of now. Could you please give me more information about the technologies used for this project and if it is feasible to add to it as a beginner? I thought of starting a new random data generator module, or help on an existing one. (in any case for the course i need to open a new issue and not work on an existing one). Would love your insights on that! Thanks

**Comment by ST-DDT** [comment, marked as answer]
The following section is my personal opinion and does not represent the general opinion of FakerJs:

> I am a Computer Science student looking for an OSS to contribute to, for a course assignment.
> in any case for the course i need to open a new issue and not work on an existing one

Please tell your teacher, that these kind of exercises tend to not benefit the OSS community.
Especially with the `not work on an existing one` limitation.
As they force you to contribute to a project that you are not familiar with and thus take additional time from the maintainers to answer basic/unrelated questions or do extra work to make a simple change merge-able.
This may benefit the student, but at the cost of the (unpaid/underpaid) community.
Here is a video that goes into more details: https://www.youtube.com/watch?v=5nY_cy8zcO4 
(For a maintainer it doesn't matter, if you do it for your CV, a school exercise or Hacktoberfest)
You may send your teacher to this comment, if they have further questions.

I personally would recommend the following rule on **when to contribute**:

- If you encountered a bug while using the product
  - A new **issue with reproduction steps** is a very valuable contribution
- If you **need** a feature for your project and not just think it might be useful to somebody else
- If you want to give it a try and fix a known/your bug or add a requested feature:
  - please make yourself familiar with the contributing guidelines
  - and try to **use/"waste" your own time** to complete the PR instead of the maintainers time.

---

> Could you please give me more information about the technologies used for this project

Please have a look at the package.json
The easiest and most effective way to learn about the technologies is looking them up using your favorite online search or AI assistant.

> if it is feasible to add to it as a beginner?

Sure, we love contributions.
Especially adding missing data to foreign locales.
New features require a certain amount of community interest before they are accepted.

See also:
- https://github.com/faker-js/faker/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22
- https://goodfirstissue.dev/

**Comment by matthewmayer** [comment]
Some easy ways you could get started would be:

1. Make a tutorial and post it somewhere, for example I wrote https://dev.to/matthewmayer/leveling-up-your-custom-fake-data-with-fakerjs-3f7m - when its finished you can add it to https://github.com/faker-js/awesome-faker
 
2. Improving the documentation - browse through https://next.fakerjs.dev/api/  and look for any incorrect spelling, grammar etc. If you find anything, open a PR to fix it.

3. Add some missing locale data - Faker has lots of methods (name, phone numbers, animals, etc) and lots of locales (English, Finnish, Hungarian, Spanish, etc) but not every locale has data for every method, for example we don't have animals in Czech, we don't have phone numbers in Finnish. If there is a language/locale you know well (looks from your profile like you maybe speak Greek?) then adding or correcting some data for that language would be helpful.

Echoing what ST-DDT wrote above, try to minimize the amount of work for the maintainers. Try to seek help from your school if you get stuck.
